scipy/optimize/_differentialevolution.py

```diff
@@ -480,41 +558,82 @@ def solve(self):
-            for candidate in range(np.size(self.population, 0)):
+                
+            # brake the population to subpopulation depending on the size of the pool
+            # determine number of sub-populations 'nsp', or number of 
+            # parallel evaluations, which is 'nsp + 1'
+            # and the length of the reminder 'lenrem'
+            nsp, lenrem = divmod(np.size(self.population, 0), self.pool.poolsize())
+            
+            itsp = 0
+            # iterate among sub-populations
+            # when self.pool.poolsize() == 1 the mutation is aggressive
+            # when self.pool.poolsize() > 1 the mutation is quasi-aggressive
+            while itsp <= nsp:
+                # determine the length of the subpopulation
+                lensp = self.pool.poolsize() if (itsp < nsp) else lenrem
+
+                # initialize list for all members (parameters) of the current subpopulation
+                spparams = []
+                # mutate parameters for the sub-population
+                for candidate in xrange(lensp):
+                    trial = self._mutate(candidate + self.pool.poolsize()*itsp)
+                    self._ensure_constraint(trial)
+                    spparams.append(self._scale_parameters(trial), *(self.args))
+                    nfev += 1
+                    
+                # in parallel case the self.func must return a list of energies
+                # for the whole subpopulation
+                spenergies = self.pool.map(self.func, spparams)
+
+                # check the number of evaluation of the functions
+                # (this, perhaps, should be deprecated as unnecessary)
                 if nfev > self.maxfun:
                     warning_flag = True
                     status_message = _status_message['maxfev']
                     break
 
-                trial = self._mutate(candidate)
-                self._ensure_constraint(trial)
-                parameters = self._scale_parameters(trial)
-
-                energy = self.func(parameters, *self.args)
-                nfev += 1
-
-                if energy < self.population_energies[candidate]:
-                    self.population[candidate] = trial
-                    self.population_energies[candidate] = energy
-
-                    if energy < self.population_energies[0]:
-                        self.population_energies[0] = energy
-                        self.population[0] = trial
-
-            # stop when the fractional s.d. of the population is less than tol
-            # of the mean energy
-            convergence = (np.std(self.population_energies) /
-                           np.abs(np.mean(self.population_energies) +
-                                  _MACHEPS))
-
+                # update population and their energies if subpopulation members are 
+                # better by iteration among all members (or jobs) of the subpopulation
+                for itjob in xrange(lensp):
```

Comment:
Fixed in https://github.com/pavelponomarev/scipy/commit/d1591124a494da81d87bbaee62d687ffa55c59b6